README.md

```diff
@@ -104,6 +104,59 @@ Fine-tuned instruction-following models are: the Code Llama - Instruct models `C
 Code Llama is a new technology that carries potential risks with use. Testing conducted to date has not — and could not — cover all scenarios.
 In order to help developers address these risks, we have created the [Responsible Use Guide](https://github.com/facebookresearch/llama/blob/main/Responsible-Use-Guide.pdf). More details can be found in our research papers as well.
 
+### Parallelization
+
+CodeLLama is built on top of [FairScale](https://github.com/facebookresearch/fairscale), a PyTorch extension library that extends and new SOTA scaling techniques.
+As explained in [Inference section](#Inference), CodeLlama requires model-parallelization (MP) for CodeLlama-13B and CodeLlama-34B.
+CodeLlama also supports other parallelization mechanisms provided by FairScale, notably, [Pipeline Parallelism (PP)](https://fairscale.readthedocs.io/en/latest/deep_dive/pipeline_parallelism.html) and Distributed Data Parallelism (DDP).
+
+
+
+#### Pipeline parallelism (PP)
+
+Examples using `CodeLlama-7b-Instruct` on two GPUs using PP:
+```
+torchrun --nproc_per_node 2 example_instructions.py \
+    --ckpt_dir CodeLlama-7b-Instruct/ \
+    --tokenizer_path CodeLlama-7b-Instruct/tokenizer.model \
+    --max_seq_len 512 --max_batch_size 4 --pipeline_length 2
+```
```

Comment:
I believe that your change works, but we decided against adding new features in this repository. The reason is that we would then need to ensure that the model performs similarly with pipeline parallelism.
Feel free to keep this in a branch or to clone our repository to support it.

Reply:
Hi @brozi, Ok! I understand. From my side, I will check the performance of the model using pipeline parallelism.